### PR TITLE
Collective lift and shift updates

### PR DESCRIPTION
**Pull request by jonrandahl**
This PR includes and resolves the following updates:

- Updated the README to improve the clarity of the instructions for
  running the application locally, as well as releasing a new version.
- Updated and improved the build files for the new infrastructure use.
- Minor text changes to the `Gemfile` to include instructions for running
  Epimorphics specific gems locally during the development of those gems.
- Updated the production `json_rails_logger` gem version to be at least the
  current version `~>0.3.5` (this is to cover out of sync release versions)
- Updated the production `lr_common_styles` gem version to be at least the
  current version `~>1.9.1` (this is to cover out of sync release versions)
- Refactored the version cadence creation to include a SUFFIX value if
  provided; otherwise no SUFFIX is included in the version number.
- Renamed the global env variable `RAILS_RELATIVE_URL_ROOT` to
  `APPLICATION_ROOT` to be more clear on it's use in the `entrypoint.sh` code.

### CODE REVIEW

**Review by bogdanadrianmarc — APPROVED** [review]
Lgtm!

File: Gemfile [hunk 1]
```diff
@@ -31,12 +32,17 @@ group :development do
...
 gem 'qonsole-rails', git: 'https://github.com/epimorphics/qonsole-rails'
-# gem 'qonsole-rails', path: '/home/ian/projects/epimorphics/qonsole-rails'
+# gem 'qonsole-rails', path: '~/Epimorphics/clients/land-registry/projects/qonsole-rails'
+
+# TODO: For running the app locally for testing you can set this to your local path
+# gem 'json_rails_logger', '~> 0.3.5', path: '~/Epimorphics/shared/json-rails-logger/'
+# gem 'lr_common_styles', '~> 1.9.1', path: '~/Epimorphics/clients/land-registry/projects/lr_common_styles/'
+# rubocop:enable Layout/LineLength
 
+# TODO: In production you want to set this to the gem from the epimorphics package repo
```
Comment (joescottdave) [inline Gemfile:44]: Is it possible to divide a Gemfile up by environment? @jonrandahl

**Review by joescottdave — COMMENTED** [review]
All good from my perspective.

File: Gemfile (hunk 1 above)
Comment (jonrandahl) [reply Gemfile:44]: @joescottdave It is, and I did think of this, however usually as we're only changing to the "local" version of the gems while we're developing those gems locally and then testing in the dependent apps I felt this was the more appropriate approach.

**Review by der — APPROVED** [review]
Looks fine to me (as far as I can follow the rubyisms :))

**Pull request merged**